docs/howto/set-up-snapcraft.rst

```diff
@@ -0,0 +1,208 @@
+macOS, and Windows.
+
+Before you begin installation and setup, make sure your system meets the
+:ref:`system requirements <system-requirements>`.
+
+
+Install the main components
+---------------------------
+
+A working Snapcraft setup has two components -- Snapcraft itself, and a build
+provider.
+
+
+Install Snapcraft
+~~~~~~~~~~~~~~~~~
+
+.. tabs::
+
+  .. group-tab:: Linux
+
+    If your Linux distribution has snapd installed, the easiest way to install
+    Snapcraft is from the Snap Store:
+
+    .. code:: bash
+
+      sudo snap install snapcraft --classic
+
+    If your distribution lacks snapd or access to the Snap Store, but has
+    access to the Ubuntu or Debian package repository, you can install
+    Snapcraft as a deb package:
+
+    .. code:: bash
+
+      sudo apt install snapcraft
+
+  .. group-tab:: macOS
+
+    #. `Install Brew <https://brew.sh#install>`_.
+    #. Install Snapcraft:
+
+       .. code:: bash
+
+        brew install snapcraft
+
+  .. group-tab:: Windows
+
+    #. `Install WSL2 with Ubuntu 20.04 or higher
+       <https://learn.microsoft.com/en-us/windows/wsl/install#install-wsl-command>`_.
+    #. In WSL2, install Snapcraft:
+
+       .. code:: bash
+
+         sudo snap install snapcraft
+
+
+Install a build provider
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+Snapcraft is compatible with two build providers -- Multipass and LXD. It
+relies on a build provider to host an isolated build environment, like a
+sandbox. Inside this environment, software can be built and packaged as snaps
+without making potentially destructive changes to the host system.
+
+For **core22 and higher**, LXD is the default provider on Linux, and Multipass
+is the default on macOS and Windows.
+
+For **core20**, Multipass is the default provider on all platforms.
+
+
+Chose a build provider
+^^^^^^^^^^^^^^^^^^^^^^
+
+Before proceeding, you must choose a build provider.
+
+`Multipass <https://multipass.run>`_ creates and manages virtual machine (VM)
+build instances. It automates setup and teardown of cloud-style Ubuntu VMs. It
+can't be used reliably on platforms that don't support nested virtualization.
+For instance, it most likely won't itself run inside a VM.
+
+`LXD <https://linuxcontainers.org/lxd/introduction>`_ creates and manages Linux
+container images. It can operate inside VMs.
+
+If the default build provider isn't fit for your snap, you can
+:ref:`switch between them <select-a-build-provider>`.
+
+
+Install Multipass
+^^^^^^^^^^^^^^^^^
+
+If Multipass isn't installed, on first run Snapcraft will ask you'd like to
+automatically install and configure it. If you agree, it will walk you through
+installation.
+
+.. If Multipass isn't installed while running in a non-interactive mode
+.. (running from a CI/CD pipeline), snapcraft will log an error and exit.
+
+To install Multipass on its own, run:
+
+.. code:: bash
+
+  sudo snap install multipass
+
+.. tip::
+
+  With core20 snaps and Multipass, the default virtual machine is assigned 2
```

Comment:
Is it worth pointing out that this is hardcoded (and thus doesn't work) in core22+?

Reply:
The qualifier at the start should cover that.